google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/DownloadCallable.java

```diff
@@ -16,4 +16,64 @@
 
 package com.google.cloud.storage.transfermanager;
 
-public class DownloadCallable {}
+import com.google.cloud.ReadChannel;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageException;
+import com.google.common.io.ByteStreams;
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.concurrent.Callable;
+
+public class DownloadCallable implements Callable<DownloadResult> {
+  private final TransferManagerConfig transferManagerConfig;
+  private final BlobInfo originalBlob;
+
+  private final ParallelDownloadConfig parallelDownloadConfig;
+
+  public DownloadCallable(
+      TransferManagerConfig transferManagerConfig,
+      BlobInfo originalBlob,
+      ParallelDownloadConfig parallelDownloadConfig) {
+    this.transferManagerConfig = transferManagerConfig;
+    this.originalBlob = originalBlob;
+    this.parallelDownloadConfig = parallelDownloadConfig;
+  }
+
+  @Override
+  public DownloadResult call() throws Exception {
+    // TODO: Check for chunking
```

Comment:
What's the plan here? To check `setAllowChunking`, and decide whether to download byte stream ?